includes/admin/class-wc-payments-bnpl-announcement.php

```diff
@@ -0,0 +1,238 @@
+ *
+ * @package WooCommerce\Payments\Admin
+ */
+
+/**
+ * WC_Payments_Bnpl_Announcement class.
+ */
+class WC_Payments_Bnpl_Announcement {
+
+	/**
+	 * WC_Payment_Gateway_WCPay.
+	 *
+	 * @var WC_Payment_Gateway_WCPay
+	 */
+	private $gateway;
+
+	/**
+	 * WC_Payments_Account.
+	 *
+	 * @var WC_Payments_Account
+	 */
+	private $account;
+
+	/**
+	 * The current time. Useful for mocking in tests.
+	 *
+	 * @var int
+	 */
+	private $current_time;
+
+	/**
+	 * Initialize class actions.
+	 *
+	 * @param WC_Payment_Gateway_WCPay $gateway Payment Gateway.
+	 * @param WC_Payments_Account      $account Account information.
+	 * @param int                      $current_time The current time.
+	 */
+	public function __construct( WC_Payment_Gateway_WCPay $gateway, WC_Payments_Account $account, int $current_time ) {
+		$this->gateway      = $gateway;
+		$this->account      = $account;
+		$this->current_time = $current_time;
+	}
+
+	/**
+	 * Initializes this class's WP hooks.
+	 *
+	 * @return void
+	 */
+	public function init_hooks() {
+		add_action( 'current_screen', [ $this, 'maybe_enqueue_scripts' ] );
+	}
+
+	/**
+	 * Needs to run after `current_screen`, to determine which page we're currently on.
+	 *
+	 * @return void
+	 */
+	public function maybe_enqueue_scripts() {
+		if ( ! is_admin() ) {
+			return;
+		}
+
+		// Only shown once to each Administrator and Shop Manager users.
+		if ( ! current_user_can( 'manage_woocommerce' ) ) {
+			return;
+		}
+
+		// Time boxed - Campaign expires after 90 days.
+		if ( $this->current_time > strtotime( '2024-07-15 00:00:00' ) ) {
+			return;
+		}
+
+		// Only displayed to BNPL eligible countries - AU, NZ, US, AT, BE, CA, CZ, DK, FI, FR, DE, GR, IE, IT, NO, PL, PT, ES, SE, CH, NL, UK, US.
+		if ( ! in_array(
+			$this->account->get_account_country(),
+			[
+				\WCPay\Constants\Country_Code::AUSTRALIA,
+				\WCPay\Constants\Country_Code::AUSTRIA,
+				\WCPay\Constants\Country_Code::NEW_ZEALAND,
+				\WCPay\Constants\Country_Code::UNITED_STATES,
+				\WCPay\Constants\Country_Code::BELGIUM,
+				\WCPay\Constants\Country_Code::CANADA,
+				\WCPay\Constants\Country_Code::CZECHIA,
+				\WCPay\Constants\Country_Code::DENMARK,
+				\WCPay\Constants\Country_Code::FINLAND,
+				\WCPay\Constants\Country_Code::FRANCE,
+				\WCPay\Constants\Country_Code::GERMANY,
+				\WCPay\Constants\Country_Code::GREECE,
+				\WCPay\Constants\Country_Code::IRELAND,
+				\WCPay\Constants\Country_Code::ITALY,
+				\WCPay\Constants\Country_Code::NORWAY,
+				\WCPay\Constants\Country_Code::POLAND,
+				\WCPay\Constants\Country_Code::PORTUGAL,
+				\WCPay\Constants\Country_Code::SPAIN,
+				\WCPay\Constants\Country_Code::SWEDEN,
+				\WCPay\Constants\Country_Code::SWITZERLAND,
+				\WCPay\Constants\Country_Code::NETHERLANDS,
+				\WCPay\Constants\Country_Code::UNITED_KINGDOM,
+			],
+			true
+		) ) {
+			return;
+		}
+
+		if ( get_user_meta( get_current_user_id(), '_wcpay_bnpl_april15_viewed', true ) === '1' ) {
+			return;
+		}
```

Comment:
Agreed, though just FYI, I initially replaced the `get_account_country()` function that checks the account cache with a potentially faster request to collect store country from WC database settings instead. However, @frosso reminded me that since this notice only appears on WooPayments pages and appears after/while other elements are loading, this cache will always be hydrated, since this account information is required by other elements on the page. Consequently, we should be able to safely assume that we will always be pulling cached data at this juncture AFAIA.

Reply:
Updated in 3338b455f9de3e39fb6ce2633891b3360611cacb.